src/scripts/main.js

```diff
@@ -1 +1,65 @@
 'use strict';
+
+const createNotification = (message, isSuccess) => {
+  const notification = document.createElement('div');
+
+  notification.className = `notification ${isSuccess ? 'success' : 'error'}`;
+  notification.dataset.qa = 'notification';
+  notification.textContent = message;
+  document.body.appendChild(notification);
+};
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handleLeftClick = (e) => {
+    resolve('First promise was resolved');
+    document.removeEventListener('click', handleLeftClick);
+  };
+
+  document.addEventListener('click', handleLeftClick);
+
+  setTimeout(() => {
+    reject(new Error());
+    document.removeEventListener('click', handleLeftClick);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  const handleClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+      document.removeEventListener('mousedown', handleClick);
+    }
+  };
+
+  document.addEventListener('mousedown', handleClick);
+});
+
+let leftClicked = false;
+let rightClicked = false;
+
+const thirdPromise = new Promise((resolve) => {
+  const handleBothClicks = (e) => {
+    if (e.button === 0) {
+      leftClicked = true;
+    }
+
+    if (e.button === 2) {
+      rightClicked = true;
+    }
+
+    if (leftClicked && rightClicked) {
```

Comment:
The logic for resolving the third promise when both left and right clicks are detected may not work as expected if the clicks are not registered in quick succession. Consider resetting the `leftClicked` and `rightClicked` flags after resolving the promise to ensure the promise can be resolved again if needed.